File: AndroidSamples/src/main/java/com/zed3/sipua/ui/BaseActivity.java
package com.zed3.sipua.ui;

import android.app.Activity;
import android.graphics.Color;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.view.WindowManager;

public abstract class BaseActivity extends Activity {

    public abstract void onActivityCreate( Bundle savedInstanceState);


    @Override
    public void onCreate( Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        onActivityCreate(savedInstanceState);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            //透明状态栏
            if(Build.VERSION.SDK_INT>= Build.VERSION_CODES.LOLLIPOP){
                View decorView = getWindow().getDecorView();
            int option = View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                    | View.SYSTEM_UI_FLAG_LAYOUT_STABLE;
            decorView.setSystemUiVisibility(option);
            getWindow().setStatusBarColor(Color.TRANSPARENT);
            }else{
                getWindow().addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
            }
//
            //透明导航栏
//            getWindow().addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION);
        }
    }
}

File: kotlindemo/src/main/java/com/android/kotlindemo/view/activity/BaseActivity.kt
package com.android.kotlindemo.view.activity

import android.graphics.Color
import android.os.Build
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.View
import android.view.WindowManager
import com.android.kotlindemo.view.IViewer

open  abstract class BaseActivity: AppCompatActivity(), IViewer {

    internal abstract  fun onActivityCreate(savedInstanceState: Bundle?)
    internal abstract fun onActivityDestory()
    internal abstract fun initView()

    override fun showProgress() {
    }

    override fun hideProgress() {
    }

    override fun dataLoadComplete(any: Any?) {

    }

    override fun moreDataLoadComplete(any: Any?) {
    }


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        onActivityCreate(savedInstanceState)
        initStatusBar();
        initView()
    }

    override fun onDestroy() {
        super.onDestroy()
        onActivityDestory()
    }

    fun initStatusBar(){
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            //透明状态栏
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                val decorView = window.decorView
                val option = View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN or View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                decorView.systemUiVisibility = option
                window.statusBarColor = Color.TRANSPARENT
            } else {
                window.addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS)
            }
        }
    }



}